File: liqp-core/src/test/java/liqp/parameterized/LiquifyNoInputTest.java
package liqp.parameterized;

import static org.assertj.core.api.Assertions.assertThat;

import junitparams.JUnitParamsRunner;
import junitparams.Parameters;
import liqp.LiquidParser;
import liqp.node.LTemplate;
import org.junit.Test;
import org.junit.runner.RunWith;

@RunWith(JUnitParamsRunner.class)
public abstract class LiquifyNoInputTest {

  private final String input;

  protected LiquifyNoInputTest(String input) {
    this.input = input;
  }

  protected LiquifyNoInputTest() {
    this.input = null;
  }

  @Parameters(method = "testParams")
  @Test
  public void run(String templateString, Object expectedResult) {
    LTemplate template = LiquidParser.newInstance().parse(templateString);
    String rendered = String.valueOf(template.render(input));

    assertThat(rendered).isEqualTo(expectedResult);
  }
}

File: liqp-core/src/test/java/liqp/parameterized/LiquifyNoInputTest.kt
package liqp.parameterized

import assertk.assertThat
import assertk.assertions.isEqualTo
import liqp.createTestParser
import liqp.parseIfNecessary
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.Parameterized

@RunWith(Parameterized::class)
abstract class LiquifyNoInputTest @JvmOverloads constructor(val templateString: String, val expectedResult: String, val inputData: String? = null) {
  @Test
  open fun run() {
    val template = createTestParser{}.parse(templateString)
    val input = try {
      inputData.parseIfNecessary()
    } catch (e: Exception) {
      inputData
    }
    val rendered = template.render(input)

    assertThat(rendered).isEqualTo(expectedResult)
  }
}
